libr/core/cmd_search.c

```diff
@@ -1804,16 +1806,23 @@ static void do_syscall_search(RCore *core, struct search_parameters *param) {
 				}
 			}
 			if ((aop.type == R_ANAL_OP_TYPE_SWI) && ret) { // && (aop.val > 10)) {
+
+				// not sure on this
+				if (!strcmp (arch, "arm") && core->anal->bits == 16) {
```

Comment:
better use core->anal->cur->arch, and maybe put this logic outside the loop unless the arch changes in the middle of the emulation to avoid unnecessary repeated checks

Reply:
you can simplify that by doing if (isThumb(core)) {.. and then move that logic outside